### PR TITLE
feature: social links functionality

### PR DESCRIPTION
**Pull request by krawczykkatarzyna**
adding social links functionality

Screen Shot 2023-09-04 at 21 21 42
Screen Shot 2023-09-04 at 23 40 26
Screen Shot 2023-09-04 at 21 21 37
Screen Shot 2023-09-05 at 00 55 53

### CODE REVIEW

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/SocialLinksItem.tsx
Comment (CzarekDryl) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/SocialLinksItem.tsx:12]: `SocialLinlsTable` => `SocialLinksTable`

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/SocialLinksItem.tsx
Comment (joanna-pagepro) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/SocialLinksItem.tsx]: Why is this added using position absolute? Also, this looks like a kebab menu that is used everywhere across the app - shouldn't it be added as a separate component that you can reuse?

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/SocialLinksItem.tsx
Comment (joanna-pagepro) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/SocialLinksItem.tsx:44]: You don't need last:mb-0 here

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/SocialLinksItem.tsx
Comment (joanna-pagepro) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/SocialLinksItem.tsx:59]: You don't need any margin class here

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/SocialLinksItem.tsx
Comment (joanna-pagepro) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/SocialLinksItem.tsx:53]: Is this span needed? I guess that you can add all those styles to the button.

File: src/components/v5/shared/Button/types.ts
Comment (joanna-pagepro) [inline src/components/v5/shared/Button/types.ts:49]: How can a button be selected? It doesn't have such state.

**Review by joanna-pagepro — CHANGES_REQUESTED** [review]

File: src/components/v5/common/ActionSidebar/partials/EditColonyDetailsForm/hooks.ts
Comment (krawczykkatarzyna) [reply src/components/v5/common/ActionSidebar/partials/EditColonyDetailsForm/hooks.ts:35]: this is field to edit colony description not social link, this is one of field which client mentioned on call

Comment (joanna-pagepro) [reply src/components/v5/common/ActionSidebar/partials/EditColonyDetailsForm/hooks.ts:35]: Then shouldn't it be added in a separate PR?

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/types.ts
Comment (krawczykkatarzyna) [reply src/components/v5/common/ActionsContent/partials/SocialLinksTable/types.ts:11]: I wanted to use specified unions but only in one case use an empty string so it's a one of solution.

File: src/components/v5/common/ActionSidebar/partials/forms/SimplePaymentForm/SimplePaymentForm.tsx
Comment (CzarekDryl) [inline src/components/v5/common/ActionSidebar/partials/forms/SimplePaymentForm/SimplePaymentForm.tsx:13]: `import TransactionTable from '../../TransactionTable/TransactionTable';` => `import TransactionTable from '../../TransactionTable';`

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/SocialLinksTable.tsx
Comment (CzarekDryl) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/SocialLinksTable.tsx:37]: border-red-400 -> border-negative-400, we don't have red-400 color in tailwind config

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx
Comment (joanna-pagepro) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx:106]: Why is this one different than others? Doesn't it have any "selected" state?

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx
Comment (joanna-pagepro) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx:114]: There's a lot of code duplication. Maybe you can create a component - list - and just pass what you need there?

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx
Comment (joanna-pagepro) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx:115]: What is this margin right added here for?

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx
Comment (joanna-pagepro) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx:214]: useSocialLinks -> useSocialLinksForm

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx
Comment (joanna-pagepro) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx:254]: Can you explain what is going on here? I don't understand this logic

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/RowField.tsx
Comment (joanna-pagepro) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/RowField.tsx:8]: What is this component added for?

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/SocialLinksModal.tsx
Comment (joanna-pagepro) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/SocialLinksModal.tsx:40]: Why are you not using a form from react-hook-form?

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/SocialLinksModal.tsx
Comment (joanna-pagepro) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/SocialLinksModal.tsx:47]: It should be a list and you can create a separate component to use here

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/SocialLinksModal.tsx
Comment (joanna-pagepro) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/SocialLinksModal.tsx:53]: (button) -> ({ type, icon, ... })

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/SocialLinksModal.tsx
Comment (joanna-pagepro) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/SocialLinksModal.tsx:80]: @username ?

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/SocialLinksModal.tsx
Comment (joanna-pagepro) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/SocialLinksModal.tsx:77]: Some styles are repeated. You can add a wrapper and apply those styles there. Also, why is this margin added here?

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/SocialLinksModal.tsx
Comment (joanna-pagepro) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/SocialLinksModal.tsx:85]: Please check if @TomaszChmielPagepro merged his PR with FormInputBase and use it here if available

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/types.ts
Comment (joanna-pagepro) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/types.ts:25]: This type shouldn't be added here. It should be add in the same directory as the component where it's used

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/types.ts
Comment (joanna-pagepro) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/types.ts:3]: Please move it to consts file

File: src/components/v5/shared/Button/Button.module.css
Comment (joanna-pagepro) [inline src/components/v5/shared/Button/Button.module.css:42]: Please add hover on desktop only

File: src/components/v5/shared/Button/Button.module.css
Comment (joanna-pagepro) [inline src/components/v5/shared/Button/Button.module.css:41]: This name is related to a place where this variant is used. Please make it more reusable.

File: src/components/v5/shared/Button/Button.tsx
Comment (joanna-pagepro) [inline src/components/v5/shared/Button/Button.tsx:31]: Please remove this prop and just use another mode

File: src/context/ActionSidebarContext.tsx
Comment (joanna-pagepro) [inline src/context/ActionSidebarContext.tsx:53]: Why is this added to the context?

**Review by joanna-pagepro — CHANGES_REQUESTED** [review]

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx
Comment (CzarekDryl) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx:117]: add hover only on desktop

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx
Comment (krawczykkatarzyna) [reply src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx:238]: for example when user created two links and want to edit one of them, the modal will be open and the correct link type will be marked and input should be filled by correct link

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/CellComponent.tsx
Comment (krawczykkatarzyna) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/CellComponent.tsx:8]: this component is used to display cell in table

File: src/components/v5/shared/Button/Button.tsx
Comment (joanna-pagepro) [reply src/components/v5/shared/Button/Button.tsx:32]: Yeah, you can create another one :)

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx
Comment (joanna-pagepro) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx:97]: I think that it can be just a const. I don't see a reason why should it stay as a hook

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/SocialLinks.tsx
Comment (joanna-pagepro) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/SocialLinks.tsx:20]: Please use styles instead of `isMobile`

File: src/components/v5/shared/Button/Button.module.css
Comment (joanna-pagepro) [reply src/components/v5/shared/Button/Button.module.css:42]: I can't see any changes here

File: src/components/v5/shared/Button/Button.module.css
Comment (joanna-pagepro) [reply src/components/v5/shared/Button/Button.module.css:41]: I can't see any changes here

**Review by joanna-pagepro — CHANGES_REQUESTED** [review]

File: src/components/v5/shared/Button/Button.tsx
Comment (krawczykkatarzyna) [reply src/components/v5/shared/Button/Button.tsx]: added new prop and fixed

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/SocialLinksTable.tsx
Comment (krawczykkatarzyna) [reply src/components/v5/common/ActionsContent/partials/SocialLinksTable/SocialLinksTable.tsx]: fixed

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx
Comment (krawczykkatarzyna) [reply src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx]: fixed

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx
Comment (krawczykkatarzyna) [reply src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx]: fixed

**Review by joanna-pagepro — APPROVED** [review]

**Review by CzarekDryl — APPROVED** [review]

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/consts.tsx
Comment (TomaszChmielPagepro) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/consts.tsx:33]: Can you create a interface/type for this?

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx
Comment (TomaszChmielPagepro) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx:102]: You can move this outside of this hook

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx
Comment (TomaszChmielPagepro) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx:111]: Are these defaults needed?

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx
Comment (TomaszChmielPagepro) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx:117]: I think you can get value from controller

Comment (TomaszChmielPagepro) [reply src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx:117]: Also is this link name or link type?

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx
Comment (TomaszChmielPagepro) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx:142]: What is this whole logic for?

File: src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx
Comment (TomaszChmielPagepro) [inline src/components/v5/common/ActionsContent/partials/SocialLinksTable/hooks.tsx:160]: Are these methods async?

**Review by rdig — CHANGES_REQUESTED** [review]
- **Please do rebase pulls, not merge pulls**
Screenshot from 2023-11-08 14-06-39
Screenshot from 2023-11-08 14-06-44
Screenshot from 2023-11-08 14-06-49
Screenshot from 2023-11-08 14-06-53
Screenshot from 2023-11-08 14-06-56
Screenshot from 2023-11-08 14-07-03
Screenshot from 2023-11-08 14-07-05
Screenshot from 2023-11-08 14-07-18

- **Please avoid duplicate commits**
Screenshot from 2023-11-08 14-07-10

- **Please be more descriptive in your commits**
Screenshot from 2023-11-08 14-06-59
Screenshot from 2023-11-08 14-07-15
Screenshot from 2023-11-08 14-07-18

- Error state message is not correct
Screenshot from 2023-11-08 14-00-49

- After clicking create action, the form resets and is not populated with data anymore. It should be populated with the colony's details

Screenshot from 2023-11-08 14-03-48

**Comment by TomaszChmielPagepro** [comment]
I have cleaned up this PR a bit can you guys review it again?

**Review by KornelKwiatkowski — DISMISSED** [review]

**Review by melyndav — CHANGES_REQUESTED** [review]
Hi @krawczykkatarzyna. Nice work! I have some changes for this. 

1. Top left icon doesn't match Figma size:
image

2. A number of issues with the all social buttons:
- Padding doesn't match Figma.
- Icon size and centre alignment doesn't match Figma
- Button text is wrong. This should be the primary button size.
- Border colour is wrong. 

image

3. Select state for the button should be text-gray-900.

image

4. The text input should be text-gray-900 
image

5. The Twitter and Instragram input fields should have the following changes:

- Change the field title to 'Twitter URL' and 'Instagram URL'
- Change the validation to URL format and not the @handle validation (same as Discord and Telegram etc) 
image
image


6. The table in the action should be titled 'External links' not social links. As per Figma.
image

7. The formatting of the text in the table under the type column should match Figma:
image

8. The colour of the text in the Username/URL field should match Figma
image


9. The column widths in the table should match Figma. 
image

10. There are no horizontal rules in any table in the actions. Only tables outside of actions.
image


11. Table header is missing border bottom as per Figma.
image

12. The menu from the table kebab icon has issues:

- Icon sizings are wrong
- The text should change on hover to a medium as per all other menus
- The 'Remove row' should be 'Delete'

Correct in Figma:
image

File: src/components/v5/common/Fields/InputBase/InputBase.tsx
Comment (joanna-pagepro) [inline src/components/v5/common/Fields/InputBase/InputBase.tsx:62]: What is this border-0 added here for?

File: src/components/v5/common/Fields/RadioButtons/TileRadioButtons/TileRadioButtons.tsx
Comment (joanna-pagepro) [inline src/components/v5/common/Fields/RadioButtons/TileRadioButtons/TileRadioButtons.tsx:38]: You have this condition added twice

**Review by joanna-pagepro — CHANGES_REQUESTED** [review]

File: src/components/v5/common/Fields/InputBase/InputBase.tsx
Comment (TomaszChmielPagepro) [reply src/components/v5/common/Fields/InputBase/InputBase.tsx:62]: Fixed

File: src/components/v5/common/Fields/RadioButtons/TileRadioButtons/TileRadioButtons.tsx
Comment (TomaszChmielPagepro) [reply src/components/v5/common/Fields/RadioButtons/TileRadioButtons/TileRadioButtons.tsx:38]: Fixed

**Comment by TomaszChmielPagepro** [comment]
@melyndav Thanks for the feedback, I have fixed all the issues.

Regarding 12. icons size seems to be fine (it's 16px on the design and 16px in the browser as well)

**Review by arrenv — CHANGES_REQUESTED** [review]
I am seeing this when trying to save the details.

image

**Review by melyndav — COMMENTED** [review]
Hey @TomaszChmielPagepro thanks for making these changes. A couple of issues:

1. Font weight in the username/URL table should be 400 it's currently 500.
2. In the kebab dropdown, the 'remove' should be 'Delete' as per Figma. 
3. Can you remove the word 'UserName' from the coloum title and just leave URL.

image

4. The kebab icon should stay text-blue-400 when active and the dropdown menu is showing - it currently defaults back to the lighter gray. 

image


5. Cancel button border in the model doesn't match Figma colour.

image

**Comment by TomaszChmielPagepro** [comment]
@arrenv @melyndav Issues fixed

**Review by KornelKwiatkowski — DISMISSED** [review]

**Review by arrenv — CHANGES_REQUESTED** [review]
Can this please be rebased on `master` it looks like it is missing some updates that means I am not able to access it locally.

**Comment by TomaszChmielPagepro** [comment]
@arrenv rebased on `master`

**Review by melyndav — APPROVED** [review]
Hey @TomaszChmielPagepro thanks for all your work on the updates, this one is approved.

**Review by joanna-pagepro — DISMISSED** [review]

**Review by arrenv — APPROVED** [review]
This is looking correct now to me, thank you!

**Comment by TomaszChmielPagepro** [comment]
@rdig can you please take a look?

**Comment by rdig** [comment]
@arrenv and @melyndav already approved it, and I trust their decision, so this can be merged

**Comment by TomaszChmielPagepro** [comment]
@rdig I am pinging you just because you have requested some changes before that's why I need your approval 😅
image

**Comment by rdig** [comment]
@thomassdk Done

**Comment by rdig** [comment]
@TomaszChmielPagepro done

**Review by ArmandoGraterol — CHANGES_REQUESTED** [review]
Good work! Quite the chunk of code.

I've noticed that when the social link is too long, it overflows (See Github link):
FireShot Capture 215 - Extensions - Colony - - localhost

also, in the same screenshot. The validation is not allowing me to submit, telling me that the colony name is required? Even thought I wrote a different name than the current one. It shouldn't be required anyways, right? We should be able to change any detail of the colony without the other details being required (In this case, I just wanted to change the social links)

**Comment by TomaszChmielPagepro** [comment]
@ArmandoGraterol Changes added

**Review by ArmandoGraterol — COMMENTED** [review]
Thank you for the quick changes and for going back and updating the other imports for form validation.

I'm still seeing that the long links are overflowing and going over the hamburger menu:

FireShot Capture 224 - Extensions - Colony - - localhost

**Review by melyndav — COMMENTED** [review]
Hey @TomaszChmielPagepro you can see the truncation of URLs here in Figma is this helps - https://www.figma.com/file/5V8pr7iMwXsT9L3VAZsmUt/Colony-V5?type=design&node-id=15705-287130&mode=design&t=AM3pW3W3YztIzFZp-4

**Comment by TomaszChmielPagepro** [comment]
Yeah, sorry missed that one, updated

**Review by ArmandoGraterol — APPROVED** [review]
LGTM 👌 Thank you for the quick changes!

**Review by joanna-pagepro — APPROVED** [review]

**Pull request merged**
